File: engine/src/com/magma/engine/assets/MapLoader.java
package com.magma.engine.assets;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.maps.tiled.TiledMap;
import com.badlogic.gdx.maps.tiled.TmxMapLoader;
import com.badlogic.gdx.utils.Array;
import com.magma.engine.MagmaGame;
import com.magma.engine.maps.MapStage;

public class MapLoader {
	
	private static Array<MapStage> maps;
		
	public static TiledMap loadTilemap(String mapName) {
		// autofill mapPath handle
		mapName = MagmaGame.getAssetPrefix() + "maps/" + mapName + ".tmx";
		Gdx.app.log("MapStage", "Loading map " + mapName);
		
		// check if file exists
        if (!Gdx.files.internal(mapName).exists()) {
			throw new IllegalArgumentException("Map " + mapName + " does not exist!");
		}
		
		try {
			TiledMap map = new TmxMapLoader().load(mapName);
			return map;
		} catch (Exception e) {
			throw e;
		}
	}
	
	public static MapStage getRegisteredMap(int id) {
		if (id > maps.size || id < 0) {
			throw new IllegalArgumentException("No registered map with id " + id + " (total maps " + maps.size + ")");
		}
		return maps.get(id-1);
	}
}

File: engine/src/com/magma/engine/assets/MapLoader.kt
package com.magma.engine.assets

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.maps.tiled.TiledMap
import com.badlogic.gdx.maps.tiled.TmxMapLoader
import com.badlogic.gdx.utils.Array
import com.magma.engine.MagmaGame
import com.magma.engine.maps.MapStage

object MapLoader {
    private val maps: Array<MapStage> = Array()

	fun loadTilemap(mapName: String): TiledMap {
        // autofill mapPath handle
        val fullPath = MagmaGame.assetFolder + "maps/" + mapName + ".tmx"
        Gdx.app.log("MapStage", "Loading map $fullPath")

        // check if file exists
        require(Gdx.files.internal(fullPath).exists()) { "Map $fullPath does not exist!" }
        return TmxMapLoader().load(fullPath)
    }

    fun getRegisteredMap(id: Int): MapStage {
        require(!(id > maps.size || id < 0)) { "No registered map with id " + id + " (total maps " + maps.size + ")" }
        return maps[id - 1]
    }
}